Add jazzy documentation and example project (#93)
JamitLabs/Jetworking: Documentation/README.md

# Documentation

This folder organizes everything needed to generate the [documentation of `Jetworking`](https://jamitlabs.github.io/Jetworking/):

- `Guides`: Markdown files to be included in the generated documentation.

> - `Jazzy Configurations`: Used to configure the documentation generation engine.

Reviewer: ```suggestion
- `Guides`: markdown files to be included in the generated documentation.
- `Jazzy Configurations`: configuration files of documentation generation engine.
```

> The actual generated documentation files are stored in the top-level `docs` folder.

Reviewer: ```suggestion
The generated documentation files are stored in the top-level `docs` folder.
```

## Regenerating documentation

> Whenever, the documentation of a `public`or `open` interface is adjusted, the documentation shall be updated. This can be done by running `make generate-docs` from the root directory of the repository (not: from this directory).

Reviewer: ```suggestion
Whenever the specification of a `public` or `open` interface changes, the documentation can be easily updated by running `make generate-docs` from the [root directory](../).
```

> To generate the docs, [Jazzy](https://github.com/realm/jazzy) is required, which can be installed as follows:

Reviewer: ```suggestion
For the docs generation [Jazzy](https://github.com/realm/jazzy) is used, which can be installed using RubyGems as follows:
```

```
sudo gem install jazzy
```

## Adding documentation for a new sub module

> When a new sub module is added, is is also required to update the documentation generation mechanism:

Reviewer: ```suggestion
When a new submodule is added, the documentation generation configuration should also be updated:
```

1. Create the `Guides/SubGuides/NewModuleName` folder with a file `Simple Sample Usage.md` in it (analogously to the way it is done for the `DataTransfer` sub module).
2. In the `Guides/MainGuide.md` file, add the new sub module to the list of available references.

> 3. Create a new jazzy configuration `NewModuleName.yaml` in the `Jazzy Configurations` folder. It may sensible to just copy and adjust `DataTransfer.yaml` for this.

Reviewer: ```suggestion
1. Create the `Guides/SubGuides/NewModuleName` folder with a file `Simple Sample Usage.md` in it (analogously how it is done in the `DataTransfer` submodule).
2. In the `Guides/MainGuide.md` file, add the new submodule to the list of available references.
3. Create a new jazzy configuration `NewModuleName.yaml` in the `Jazzy Configurations` folder (you could just copy `DataTransfer.yaml` and edit accordingly).
```
